### PR TITLE
Music bar song scrolling

### PR DESCRIPTION
**Pull request by gordon-cheung**

### CODE REVIEW

File: SDL/src/MusicBar.cpp [hunk 1]
```diff
@@ -58,15 +62,99 @@ void MusicBar::drawSongName()
...
+        }
+    } 
+    else 
+    {
+        songLocation = {(500) - (songWidth/2), musicbarSurfaceHeight/2 - songHeight/2, 0, 0}; //880 is start of volume bar, 50 is end of current time
+        SDL_BlitSurface(songSurface, NULL, surface, &songLocation);
+        SDL_FreeSurface(songSurface);
+    }
+    
+    SDL_Surface *Rect1Surface;
```
Comment (mpminardi) [inline SDL/src/MusicBar.cpp]: This should be camel case

File: SDL/src/MusicBar.cpp [hunk 2]
```diff
@@ -58,15 +62,99 @@ void MusicBar::drawSongName()
...
+    Rect1Surface = SDL_CreateRGBSurface(0, 150, 64, 32, 0, 0, 0, 0);
+    SDL_FillRect(Rect1Surface, NULL, SDL_MapRGB(Rect1Surface->format,43,43,43));
+    SDL_Rect Rect1Location = {0,0,0,0};
```
Comment (mpminardi) [inline SDL/src/MusicBar.cpp]: Should be camel case

File: SDL/src/MusicBar.cpp [hunk 3]
```diff
@@ -58,15 +62,99 @@ void MusicBar::drawSongName()
...
+    SDL_BlitSurface(Rect1Surface, NULL, surface, &Rect1Location);
+    SDL_FreeSurface(Rect1Surface);
+
+    SDL_Surface *Rect2Surface;
```
Comment (mpminardi) [inline SDL/src/MusicBar.cpp]: Same as above

**Comment by mpminardi** [comment]
We should really move away from using magic numbers everywhere. Having defined constants will likely save us headaches in the future.

File: SDL/src/MusicBar.cpp [hunk 4]
```diff
@@ -58,15 +62,99 @@ void MusicBar::drawSongName()
     
     // SDL CALLS // 
     
+   // SDL_Surface *songSurface;
+    //SDL_Color songColor = {255, 255, 255}; 
+    //songSurface = TTF_RenderUTF8_Blended(songNameFont, songChar, songColor);
+    //TTF_SizeText(songNameFont, songChar, &songWidth, &songHeight);
+    //SDL_Rect songLocation = {(880-50)/2 - songWidth/2, musicbarSurfaceHeight/2 - songHeight/2, 0, 0}; //880 is start of volume bar, 50 is end of current time
+    //SDL_BlitSurface(songSurface, NULL, surface, &songLocation);
+    //SDL_FreeSurface(songSurface);
```
Comment (mpminardi) [inline SDL/src/MusicBar.cpp]: Can the commented out section above be removed?

**Comment by gordon-cheung** [comment]
I made updates to the code, please review!

**Comment by bseto** [comment]
Tested. Lengthy code that could be made into smaller functions. This will have to be cleaned up later, we can get someone new to learn by cleaning it up:)

Looks good Gordon xD

**Pull request merged**